service-print/src/main/java/org/oskari/print/loader/CommandLoadImageWMTS.java

```diff
@@ -190,11 +195,44 @@ private TileMatrix getTileMatrix(TileMatrixSet tms) throws IllegalArgumentExcept
     }
 
     private TileMatrixSet getTileMatrixSet() throws IllegalArgumentException {
+        List<TileMatrixSet> possibleTileMatrixSets = new ArrayList<>();
+
         for (TileMatrixSet tms : capabilities.getTileMatrixSets()) {
             if (srs.equals(ProjectionHelper.shortSyntaxEpsg(tms.getCrs()))) {
-                return tms;
+                possibleTileMatrixSets.add(tms);
+            }
+        }
+
+        if (possibleTileMatrixSets.isEmpty()) {
+            throw new IllegalArgumentException("Could not find TileMatrixSet for the requested crs");
+        }
+
+        if (possibleTileMatrixSets.size() == 1) {
+            return possibleTileMatrixSets.get(0);
+        }
+        return determineTileMatrixSetToUse(possibleTileMatrixSets);
+    }
+
+    private TileMatrixSet determineTileMatrixSetToUse(List<TileMatrixSet> possibleTileMatrixSets) throws IllegalArgumentException {
+        JSONObject useThisTileMatrixSetInstead = layerService.find(layer.getId()).getOptions().optJSONObject("useThisAlternativeTileMatrixSet");
+        if (useThisTileMatrixSetInstead == null) {
+            return possibleTileMatrixSets.get(0);
+        } 
+        
+        TileMatrixSet setToReturn = null;
+        for (TileMatrixSet tms : possibleTileMatrixSets) {
```

Comment:
This loop is still doing weird things.
This is what I had in mind:
```
String id = useThisTileMatrixSetInstead.optString(srs);
if (id == null) {
    return possibleTileMatrixSets.get(0);
}
for (TileMatrixSet tms : possibleTileMatrixSets) {
    if (tms.getId().equals(id)) {
        return tms;
    }
}
throw new IllegalArgumentException("Could not find TileMatrixSet with id " + id + " for layer " + layer.getId());
```
Or if you prefer Streams then instead of the foor loop:
```
return possibleTileMatrixSets.stream()
        .filter(tms -> tms.getId().equals(id))
        .findAny()
        .orElseThrow(() -> new IllegalArgumentException("Could not find TileMatrixSet with id " + id + " for layer " + layer.getId()));

Reply:
I thing I like loops more.
They seem more readable.

Oh... 
Well then I had misunderstanding.
Since you said: 

> ... TileMatrixSet identifiers as values, e.g. "useThisAlternativeTileMatrixSet":{ "EPSG:3879": "ETRS-GK25", "EPSG:3880": "ETRS-GK26" }

I thought that for some reason all of them should be looked to know if any of them is available.

Reply:
Yea the idea is that if the PrintRequest is for example in EPSG:3067 then we'd use TileMatrixSet with id "ETRS-TM35FIN" or if the request is in EPSG:3879 we'd use TileMatrixSet with id "ETRS-GK25" etc. That way the same OskariLayer can be used regardless of the current projection.